core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/SystemSettingsCredentialsProvider.java

```diff
@@ -67,8 +86,16 @@ public AwsCredentials resolveCredentials() {
                                     .build();
         }
 
-        return sessionToken == null ? AwsBasicCredentials.create(accessKey, secretKey)
-                                    : AwsSessionCredentials.create(accessKey, secretKey, sessionToken);
+        if (webIdentityTokenFile != null && roleArn != null) {
```

Comment:
Should we warn if one is specified without the other?

Reply:
I'd be fine warning if web identity was specified and role arn wasn't but not the other way around.